src/clients/topUpCredit.js

```diff
@@ -0,0 +1,32 @@
+import Credit from "../models/credit.js";
+
+export default async (creditAmount) => {
+
+  try {
+    const existingCredit = await Credit.findOne()
```

Comment:
This doesn't make much sense without some type of query. I will always return the first element in the collection (or a random one, not sure on the behaviour right now).

Also if you're going to find and then update you can use findAndUpdate